### PR TITLE
Add exceptions and bugfixes

### PR DESCRIPTION
**Pull request by hiongkaihan**

### CODE REVIEW

File: src/test/java/seedu/duke/DeleteCommandTest.java [hunk 1]
```diff
@@ -118,4 +118,37 @@ public void executeDeleteCommand_recordDoesNotExist() {
...
+        data.loadCurrentPatient(patient.getID());
+        HashMap<String, String> arguments = new HashMap<>();
+        arguments.put("command", "delete");
+        arguments.put("r", "");
+        Ui ui = new Ui();
+        DeleteCommand deleteCommand = new DeleteCommand(ui, data, arguments);
+
+        Exception exception = assertThrows(Exception.class, () -> {
+            deleteCommand.execute();
+        });
```
Comment (fsgmhoward) [inline src/test/java/seedu/duke/DeleteCommandTest.java:151]: For the `Exception` here, you could just change it to `InvalidInputException` so any other ones (unexpected ones) will make it fail.

File: src/test/java/seedu/duke/AddCommandTest.java [hunk 2]
```diff
@@ -49,6 +49,36 @@ public void executeAddCommand_invalidID_exceptionThrown() {
...
+        Data data = new Data();
+        Ui ui = new Ui();
+        HashMap<String, String> arguments = new HashMap<>();
+        arguments.put("command", "add");
+        arguments.put("payload", "S1234567A");
+        AddCommand addCommand = new AddCommand(ui, data, arguments);
+
+        Exception exception = assertThrows(Exception.class, () -> {
+            addCommand.execute();
+        });
```
Comment (fsgmhoward) [inline src/test/java/seedu/duke/AddCommandTest.java:78]: Same here. See the other comment.

**Review by fsgmhoward — APPROVED** [review]
Looks good to me 👍  Just a few very minor points that you might want to change before merging.

File: src/main/java/seedu/duke/command/DeleteCommand.java [hunk 3]
```diff
@@ -30,6 +30,9 @@ public void execute() throws InvalidInputException, StorageException {
 
         if (arguments.containsKey(Constants.PATIENT_KEY)) {
             String id = arguments.get(Constants.PATIENT_KEY);
+            if (id == "") {
```
Comment (brandonfoong) [inline src/main/java/seedu/duke/command/DeleteCommand.java:33]: Should it be `String.equals` here?

**Pull request merged**